packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart

```diff
@@ -116,4 +123,65 @@ void main() {
               message: 'It appears the output file contains an error message, not valid output.'));
     });
   });
+
+  group('Screenshot for devices unsupported for project', () {
+    late _TestDeviceManager testDeviceManager;
+
+    setUp(() {
+      testDeviceManager = _TestDeviceManager(logger: BufferLogger.test());
+    });
+
+    testUsingContext('should not throw', () async {
```

Comment:
can you also add a test for when there are multiple devices? I'm guessing it should tool exit with a message to pass the -d option.

Reply:
I have added a test for this.